rerun_py/rerun_sdk/rerun/_send_columns.py

```diff
@@ -22,9 +26,97 @@ def as_arrow_array(self) -> pa.Array:
         ...
 
 
+class IndexColumn(TimeColumnLike):
```

Comment:
Same comment I made elsewhere: I really wish we stopped introducing new python code in the python SDK, and try to do it all on the Rust side if possible (like e.g. @jleibs' dataframe APIs).